### ISSUE TITLE
dial attempt failed for localhost domain, but success for 127.0.0.1

### ISSUE THREAD

**Issue by photonxp** [ipfs/go-ipfs#3558]
#### Version information:
go-ipfs version: 0.4.4-
Repo version: 4
System version: amd64/linux
Golang version: go1.7

Chrome Version 47.0 (64-bit)

#### Type:
Bug

#### Priority:
P4

#### Description:
Chrome browser shows different response for LOCALHOST and 127.0.0.1 domain

LOCALHOST REQUEST    http://localhost:5001/api/v0/swarm/connect?arg=/ip4/149.56.14.160/tcp/4001/ipfs/QmU33iCTrgMhurhEvLG25j4AMz82r8P3wqBPwUAtKmBHXX
RESPONSE    {"Message":"connect QmU33iCTrgMhurhEvLG25j4AMz82r8P3wqBPwUAtKmBHXX failure: dial attempt failed","Code":0}

127.0.0.1 REQUEST    http://127.0.0.1:5001/api/v0/swarm/connect?arg=/ip4/149.56.14.160/tcp/4001/ipfs/QmU33iCTrgMhurhEvLG25j4AMz82r8P3wqBPwUAtKmBHXX
RESPONSE    {"Strings":["connect QmU33iCTrgMhurhEvLG25j4AMz82r8P3wqBPwUAtKmBHXX success"]}

**Comment by lgierth** [comment]
I don't think this has anything todo with "127.0.0.1"/"localhost". It's simply that swarm dials *can* fail.
